### PR TITLE
unpin netcdf4 to free and bump build number to 1

### PR DESCRIPTION
**Pull request by valeriupredoi**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

We repo patched iris to force the pin netCDF4<1.6.1 across versions, so our pin is now redundant. We have done the same in the dev repo https://github.com/ESMValGroup/ESMValCore/pull/1814

To be able to have the tests run and package be built I had to pin mypy plugin, I also released the pin on flake8 as we did in Core just recently.

### CODE REVIEW

**Comment by valeriupredoi** [comment]
@conda-forge-admin, please rerender

**Comment by valeriupredoi** [comment]
OK - I updated the environment: to be able to have the tests run and package be built I had to pin mypy plugin, I also released the pins on cf_units and flake8 as we did in Core just recently

**Comment by valeriupredoi** [comment]
mypy fails are getting a bit too hard to solve - any recommendations here guys? Or, I can just close this since we can wait for the release to get all the pins one line?

**Comment by bouweandela** [comment]
You'll probably want to pin mypy instead of the pytest plugin used to run it.

**Comment by valeriupredoi** [comment]
> You'll probably want to pin mypy instead of the pytest plugin used to run it.

I believe you are absolutely correct @bouweandela - let me try that now :+1:

**Comment by valeriupredoi** [comment]
finally :green_circle: @bouweandela can we get this in? I will open an issue with the different pins that we'll have to fix at release point after we merge (also I think we need to add @remi-kazeroni to the repo, if he's gonna be our next RM) :beer:

File: recipe/meta.yaml
Comment (valeriupredoi) [reply recipe/meta.yaml:77]: let me try remove it see what we get

Comment (valeriupredoi) [reply recipe/meta.yaml]: OK that worked :partying_face:

**Review by bouweandela — APPROVED** [review]

**Comment by valeriupredoi** [comment]
huzzah! Cheers @bouweandela :beers:

**Pull request merged**
